Dockerfile

```diff
@@ -49,27 +48,28 @@ ENV HUGGINGFACE_HUB_CACHE=/data \
 
 ARG DEBIAN_FRONTEND=noninteractive
 
-RUN curl -s https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB \
-    | gpg --dearmor | tee /usr/share/keyrings/oneapi-archive-keyring.gpg > /dev/null \
-    && echo "deb [signed-by=/usr/share/keyrings/oneapi-archive-keyring.gpg] https://apt.repos.intel.com/oneapi all main" | \
-    tee /etc/apt/sources.list.d/oneAPI.list
-
 RUN <<HEREDOC
-    apt-get update
+    apt-get update && \
     apt-get install -y --no-install-recommends \
         libomp-dev \
         ca-certificates \
         libssl-dev \
         curl \
         pkg-config \
-        # Provided for convenience when using the NCCL crate feature:
-        openmpi-bin \
-        # Provided for MKL feature runtime:
-        intel-oneapi-hpc-toolkit
+        openmpi-bin && \
 
     rm -rf /var/lib/apt/lists/*
 HEREDOC
 
+ARG WITH_FEATURES="cuda,cudnn,nccl,mkl,mpi"
```

Comment:
Since you'd like to use this ARG in the runtime stage as well to infer if the package should be installed, I would suggest hoisting the `ARG` with default assignment to the the start of the `Dockerfile`, then leave these as `ARG WITH_FEATURES` to inherit that.